### DISCUSSION TITLE
Why not using HTML files and using jQuery ?

### DISCUSSION THREAD

**Discussion by evanlegamer** [DrakenWan/Yale3#6] [Q&A] [answered]
Just little questions,

Why do you generate the html in the JS file ? And why do you use Jquery to extract content ? Maybe using pureJS with some defined function could do the work faster and without loading jquery no ?

also, i never worked on those kind of extension, how do you develop an extension ? You just modify the code with vsCode and reload the extension in chrome browser each time you do a modification ? 

Thanks in advance for your answer.

**Comment by KartikayKaul** [comment, marked as answer]
In [Rekrut ](https://github.com/drakenwan/Rekrut) extension , I was injecting an external html file using web accessible resource attribute in the manifest file. Years later when I was creating Yale3 this was not working and I was unable to inject an external file. It kept getting blocked despite all troubleshoots I tried. So I was injecting html using the contextscript file instead.

Jquery felt easier to use. It was just a personal choice and I stuck to it.

Yeah I edit my code on vscode, Save it and reload the chrome extension in `chrome://extensions` page and then see how it works on a LinkedIn page.

**Comment by KartikayKaul** [comment]
Btw we can continue this in the discussion section. I will close it here if you want.
